### ISSUE TITLE
Lua error when clicking newcomer nameplates

### ISSUE THREAD

**Issue by JeeveStobs** [Backupiseasy/ThreatPlates#457]
```
Message: .../TidyPlates_ThreatPlates/Functions/nametextcolor.lua:44: attempt to index local 'color' (a nil value)
Time: Sun Jun 11 17:40:13 2023
Count: 3
Stack: .../TidyPlates_ThreatPlates/Functions/nametextcolor.lua:44: attempt to index local 'color' (a nil value)
[string "@Interface/AddOns/TidyPlates_ThreatPlates/Functions/nametextcolor.lua"]:44: in function `SetNameColor'
[string "@Interface/AddOns/TidyPlates_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua"]:780: in function <...s_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua:778>
[string "@Interface/AddOns/TidyPlates_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua"]:848: in function <...s_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua:846>
[string "@Interface/AddOns/TidyPlates_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua"]:446: in function <...s_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua:421>
[string "@Interface/AddOns/TidyPlates_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua"]:485: in function <...s_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua:463>
[string "@Interface/AddOns/TidyPlates_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua"]:1142: in function `?'
[string "@Interface/AddOns/TidyPlates_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua"]:1096: in function <...s_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua:1095>

Locals: self = <table> {
 UIScaleChanged = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua:1876
 CheckForIncompatibleAddons = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/Core.lua:315
 Theme = <table> {
 }
 UseUniqueWidget = true
 TotemCreatureType = "Totem"
 UnitStyle_CastTrigger_Reset = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/Styles/Styles.lua:324
 UpdateExtensions = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/Widgets/TidyPlatesExtensions.lua:90
 AurasFilterMode = <table> {
 }
 IS_MAINLINE = true
 UpdateConfigurationStatusText = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/Functions/healthtext.lua:354
 SetThemes = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/Styles/init.lua:17
 ANCHOR_POINT_SETPOINT = <table> {
 }
 CheckTableStructure = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/Init.lua:342
 CVars = <table> {
 }
 THREAT_VALUE_TYPE = <table> {
 }
 InitializeCustomNameplates = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/Options.lua:395
 ImportProfile = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/Database.lua:1555
 ProfChange = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/Options.lua:10439
 SpecName = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/Core.lua:127
 GetThreatStyle = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/Styles/Styles.lua:178
 SetStyle = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/Styles/Styles.lua:335
 OpenOptions = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/Options.lua:10503
 SetThreatColor = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/Functions/threatcolor.lua:38
 MigrateDatabase = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/Database.lua:1337
 CustomPlateGetHeaderName = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/Options.lua:377
 Font = <table> {
 }
 ForceUpdate = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua:1957
 UpdateStyleForStatusText = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/Functions/healthtext.lua:342
 UnitStyle_AuraTrigger_UpdateStyle = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/Styles/Styles.lua:276
 UpdateCustomStyleIcon = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/Widgets/UniqueIconWidget.lua:189
 LEGACY_CUSTOM_NAMEPLATES = <table> {
 }
 Animations = <table> {
 }
 UnitStyle_NameDependent = <function> defined @Interface/AddOns/TidyPlates_ThreatPlates/Styl
```

Further details on WoW mentor system: https://www.wowhead.com/news/guide-mentor-system-help-new-players-starting-out-in-shadowlands-318521

Example of newcomer nameplate icon:
https://wow.zamimg.com/uploads/screenshots/normal/990491.jpg

**Comment by Backupiseasy** [comment]
The newcomer icon is the icon above the name of the player? I did not know that or have seen that before :)

It looks like the icon that is also used for GameObjects (like look icons or the wheel symbol for some NPCs). I am working on supporting that for 11.2. I'll look into this, thanks for reporting it.

**Comment by Backupiseasy** [comment]
So, first, I don't currently get this error. I also think that the icon is basically a target marker, at least after first looking into the frame stack it looks like that. Do you have any specific settings for target marker enabled/disabled?

**Comment by JeeveStobs** [comment]
The "Mentor System" is enabled on per character basis after you talk a specific NPC, details here: https://www.icy-veins.com/wow/mentor-system-overview-for-world-of-warcraft

After you opt-in and are flagged as a mentor, you see the icon, mentioned previously, for new players/newcomers.

Not sure if this can be tested on PTR.

**Comment by Backupiseasy** [comment]
I did that (on live) and it worked, i.e., I could see the icons as in your screenshot. But I did not get the Lua error from above. That's what I am trying to replicate.

**Comment by Backupiseasy** [comment]
Ok, now I can reproduce it :D

**Comment by Backupiseasy** [comment]
Fixed now in release 11.1.23

**Comment by JeeveStobs** [comment]
Confirmed fixed, thank you!
